GameServer.cpp

```diff
@@ -462,31 +462,25 @@ void GameServer::DoTimedUpdates()
 
 std::size_t GameServer::FindGameInfoClient(const sockaddr_in &from, unsigned int clientRandValue)
 {
-	// Search games list
-	for (std::size_t i = 0; i < numGames; ++i)
-	{
-		// Check if this address matches
-		if ((gameInfo[i].clientRandValue == clientRandValue) && (memcmp(&gameInfo[i].addr, &from, sizeof(sockaddr_in)) == 0))
-		{
-			// Return the GameInfo
-			return i;
-		}
-	}
-
-	// GameInfo not found
-	return InvalidGameInfoIndex;
+	return FindGameInfo(from,
+		[clientRandValue](const GameInfo& gameInfo) {return gameInfo.serverRandValue == clientRandValue; }

```

Comment:
Err, careful. The original code was: `gameInfo[i].clientRandValue == clientRandValue`

The struct has both a `clientRandValue` and `serverRandValue` field.